src/main/java/com/minecolonies/coremod/client/gui/WindowHutGuardTower.java

```diff
@@ -9,257 +10,448 @@
+                final Button switchButton = rowPane.findPaneOfTypeByID(GUI_LIST_BUTTON_SWITCH, Button.class);
 
-        buttonTaskPatrol = this.findPaneOfTypeByID(BUTTON_TASK_PATROL, Button.class);
-        buttonTaskFollow = this.findPaneOfTypeByID(BUTTON_TASK_FOLLOW, Button.class);
-        buttonTaskGuard = this.findPaneOfTypeByID(BUTTON_TASK_GUARD, Button.class);
+                if (mobsToAttack.get(index).hasAttack())
+                {
+                    switchButton.setLabel(GUI_SWITCH_ON);
+                }
+                else
+                {
+                    switchButton.setLabel(GUI_SWITCH_OFF);
+                }
+            }
+        });
+    }
 
-        buttonSetTarget = this.findPaneOfTypeByID(BUTTON_SET_TARGET, Button.class);
+    @Override
+    public void onButtonClicked(@NotNull final Button button)
+    {
+        final Pane currentPane = findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).getCurrentView();
+        if (currentPane != null)
+        {
+            final Button buttonNextPage = findPaneOfTypeByID(GUI_BUTTON_NEXT_PAGE, Button.class);
+            final Button buttonPrevPage = findPaneOfTypeByID(GUI_BUTTON_PREV_PAGE, Button.class);
+            switch (button.getID())
+            {
+                case GUI_BUTTON_NEXT_PAGE:
+                    findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).nextView();
+                    buttonPrevPage.setEnabled(true);
+                    buttonPrevPage.show();
+                    final Pane newCurrentPane = findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).getCurrentView();
+
+                    //System.out.println("Current Page:" + )
+                    if (newCurrentPane != null
+                          && newCurrentPane.getID().equals(GUI_PAGE_MOB_ACTIONS))
+                    {
+                        buttonNextPage.setEnabled(false);
+                        buttonNextPage.hide();
+                    }
+                    break;
+                case GUI_BUTTON_PREV_PAGE:
+                    findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).previousView();
+                    buttonNextPage.setEnabled(true);
+                    buttonNextPage.show();
+                    final Pane otherCurrentPane = findPaneOfTypeByID(GUI_SWITCH_VIEW_PAGES, SwitchView.class).getCurrentView();
+
+                    if (otherCurrentPane != null
+                          && otherCurrentPane.getID().equals(GUI_PAGE_PAGE_ACTIONS))
+                    {
+                        buttonPrevPage.setEnabled(false);
+                        buttonPrevPage.hide();
+                    }
+                    break;
+                default:
+                    super.onButtonClicked(button);
+                    break;
+            }
+        }
         handleButtons();
     }
 
+    @Override
+    public void onUpdate()
+    {
+        super.onUpdate();
+
+        pullInfoFromHut();
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)